skopt/searchcv.py

```diff
@@ -393,7 +393,9 @@ def add_spaces(self, names, search_spaces):
 
         if not isinstance(search_spaces, list):
             search_spaces = [search_spaces]
-        if not isinstance(names, list):
+        if isinstance(names, list):
```

Comment:
I thought originally of a use case where you can resume the calculations if necessary, or switch between calculations using `step` function, which does a single step in particular subspace which is specified as an argument to the function. This way one can have some custom stopping criterions.

Thus far I am also more leaning to reduce the complexity of the class - we can revert it when necessary.

I would suggest to do the following:

1. Merge this as is and release 0.4 soon
2. Reduce the class after the 0.4 release

There are many issues right now that people have simply because pip version of package is outdated, so I think that would be nice to fix soon.